aten/src/ATen/core/DistributionsHelper.h

```diff
@@ -68,37 +54,139 @@ template <> struct DistAccumType<double> { using type = double; };
 template <typename T>
 using dist_acctype = typename DistAccumType<T>::type;
 
-// Constants for uniform distribution
-// doubles have 52 bits of mantissa (fractional part)
-constexpr uint64_t DOUBLE_MASK = (1ULL << 53) - 1;
-constexpr double DOUBLE_DIVISOR = 1.0 / (1ULL << 53);
+/**
+ * Samples a discrete uniform distribution in the range [base, base+range) of type T
+ * This is a transformation function for
+ * https://pytorch.org/docs/stable/tensors.html?highlight=random#torch.Tensor.random_
+ * when both `from` and `to` are specified.
+ */
+template <typename T>
+struct uniform_int_from_to_distribution {
+
+  C10_HOST_DEVICE inline uniform_int_from_to_distribution(uint64_t range_, int64_t base_) {
+    range = range_;
+    base = base_;
+  }
+
+  template <typename V,
+            typename std::enable_if<std::is_integral<V>::value, int>::type = 0>
+  C10_HOST_DEVICE inline T operator()(V val) {
+    return static_cast<T>(static_cast<int64_t>((val % range) + base));
+  }
 
-// floats have 23 bits of mantissa (fractional part)
-constexpr uint32_t FLOAT_MASK = (1 << 24) - 1;
-constexpr float FLOAT_DIVISOR = 1.0f / (1 << 24);
+  template <typename RNG,
+            typename std::enable_if<(!std::is_fundamental<RNG>::value), int>::type = 0>
+  C10_HOST_DEVICE inline T operator()(RNG generator) {
+    if ((
+      std::is_same<T, int64_t>::value ||
+      std::is_same<T, double>::value ||
+      std::is_same<T, float>::value ||
```

Comment:
Do the double and float specializations actually work here?

Reply:
I have some plans to address this part but not in this PR, I'd prefer to have it as is for now for compatibility